### PR TITLE
Upgrade to django 3.1.2

### PR DESCRIPTION
**Pull request by apoorvaeternity**
## Proposed changes
- Upgrade to use django=3.1.2
- Added dependencies to `requirements.txt`
- Some code cleanup

### CODE REVIEW

**Comment by henrykironde** [comment]
Thanks for the update @apoorvaeternity

**Comment by henrykironde** [comment]
Sent you a message on hangouts

File: retrieverdash/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+pyparsing==2.4.7
+pytest==6.1.1
+pytest-django==4.0.0
+python-dateutil==2.8.1
 pytz==2017.3
-git+https://www.github.com/weecology/retriever@master
-pytest-django
-django-crontab
+requests==2.24.0
+retriever
```
Comment (henrykironde) [inline retrieverdash/requirements.txt]: I think we may still need the `dev` version here vs the `release `"git+https://www.github.com/weecology/retriever@master"

What do you think?

**Comment by henrykironde** [comment]
retrieverdash/tests/test_dashboard_script.py:13: in <module>

**Comment by henrykironde** [comment]
Am just guessing, you know better I would say, but we have this in retriever 
```
mysql_engine, postgres_engine, sqlite_engine, msaccess_engine, \
csv_engine, download_engine, json_engine, xml_engine, _ = engine_list

```
Do you think we have to add more _ for each engine not used
mysql_engine, postgres_engine, sqlite_engine, _ = engine_list

**Comment by henrykironde** [comment]
Nice 👍  `*_`

**Comment by apoorvaeternity** [comment]
Yeah - `*_` solves this. Also, I think we can have a dictionary from where we can directly refer the engine we want to use.

**Pull request merged**